colossalai/tensor/_ops/embedding.py

```diff
@@ -0,0 +1,56 @@
+import torch
+from colossalai.tensor.op_wrapper import colo_op_impl
+from colossalai.context import ParallelMode
+from colossalai.nn.layer.parallel_1d._utils import split_forward_gather_backward, reduce_input, \
+    gather_forward_split_backward, reduce_grad
+from colossalai.nn.layer.utils import divide
+from colossalai.core import global_context as gpc
+from packaging import version
+from colossalai.tensor import ComputePattern, TensorSpec, ComputePattern, ParallelAction, ColoTensor, ShardPattern
+
+def colo_embedding_1Dcol(input_tensor: ColoTensor, weight: ColoTensor, args, kwargs) -> ColoTensor:
+    # embedding_1Dcol split the weight(lookup table)
+    # Gather splitted lookup table
+    parallel_action = weight.shard_spec.get_action_by_compute_pattern(ComputePattern.TP1DCol_Embedding)
+    if input_tensor.is_activation() and not input_tensor.is_gathered():
```

Comment:
input_tensor.is_activation() is not necessary.
all input should be activation? you can use an assert to check.

Reply:
Removed this judge.